src/core/nRF52832/core.gpr

```diff
@@ -1,22 +1,45 @@
+with "switches";
 with "basalt";
+with "gneiss";
+
 project Core is
 
-   for Source_Dirs use (".");
-   for Object_Dir use "obj";
-   for Create_Missing_Dirs use "True";
-   for Languages use ("Ada", "Asm_CPP");
-   for Target use "arm-eabi";
-   for Main use ("main.adb");
+   Gneiss_Root := external ("GNEISS_ROOT", ".");
+   Step := external ("CEMENT_BUILD_STEP");
+
+   case Step is
+      when "compile" =>
+         for Source_Dirs use (Gneiss_Root & "/src/core/nRF52832");
+         for Object_Dir use external ("CEMENT_OBJECT_DIR");
+         for Create_Missing_Dirs use "True";
+         for Languages use ("Ada", "Asm_CPP");
+         for Target use "arm-eabi";
+         for Main use ("main.adb");
+         for Runtime ("Ada") use Gneiss_Root & "/ada-runtime/build/nrf52/obj";
+      when others =>
+         for Languages use ();
+   end case;
 
-   for Runtime ("Ada") use "../../../ada-runtime/build/nrf52/obj";
+   package Builder is
+      case Step is
+         when "compile" =>
+            for Global_Configuration_Pragmas use Gneiss_Root & "/spark.adc";
+            for Executable ("main.adb") use "core";
+         when others =>
+            null;
+      end case;
+   end Builder;
 
    package Compiler is
-      for Default_Switches ("Ada") use ("-gnatg", "-g", "-mthumb", "-ffunction-sections", "-fdata-sections");
+      for Default_Switches ("Ada") use Switches.Compiler_Switches & ("-g", "-mthumb", "-ffunction-sections", "-fdata-sections");
       for Default_Switches ("Asm_CPP") use ("-g", "-mthumb", "-ffunction-sections", "-fdata-sections");
    end Compiler;
 
    package Linker is
-      for Default_Switches ("Ada") use ("-Wl,--gc-sections", "-Wl,--print-memory-usage", "-T", "link.ld");
+      for Default_Switches ("Ada") use ("-Wl,--gc-sections", "-Wl,--print-memory-usage",
+                                        --  "-nostdlib", "-nostdinc",
+                                        --  "-L" & Gneiss_Root & "/ada_runtime/build/nrf52/obj/adalib",
```

Comment:
Please remove commented options

Reply:
I'd like to keep them for now. The updated runtime did not resolve all crashes and I want to keep these lines for debugging.